py/client/pydeephaven/_table_interface.py

```diff
@@ -33,7 +33,7 @@ def table_op_handler(self, table_op) -> Union[Table, Query]:
 
         ...
 
-    def drop_columns(self, cols: Union[str, List[str]]) -> Union[Table, Query]:
+    def drop_columns(self, cols: Union[str, List[str]]) -> Table:
```

Comment:
Should we be using `Generics` or something similar for these return types, instead of just picking a particular return type and using it, which is right in some cases but wrong in others?

Reply:
Unfortunately Sphinx doesn't respect concrete class derived from Generic types yet. We could move the docstrings from TableInterface into Table to avoid this confusion but it will create a lot more code as we did for the Query class.

Reply:
I would create a ticket to come back to this once Sphinx supports concrete types derived from Generic types.

1) The Union vs Table return value is not consistently applied in the file.
2) After some thought, I would consider going back to the union return type for now.  Using Table, anyone with a query is going to get bogus autocomplete and an unhappy IDE...which seems potentially worse.

Reply:
If we do union, the class docs may want to say "Methods return Table" or Methods return Query"

Reply:
https://github.com/deephaven/deephaven-core/issues/3761